lib/rig_inbound_gateway_web/vconnection.ex

```diff
@@ -0,0 +1,204 @@
+defmodule RigInboundGatewayWeb.VConnection do
+  @moduledoc """
+  The VConnection acts as an abstraction for the actual connections.
+
+  It monitors the processes, sends out heartbeats and subscription refreshes and sends out events to the serverless engine when intialized, connection is lost or connection is re-established.
+  """
+
+  alias Rig.EventFilter
+  alias Rig.RingBuffer
+  alias RigCloudEvents.CloudEvent
+  alias RigInboundGateway.Events
+
+  require Logger
+
+  use GenServer
+
+  # TODO: Discuss a reasonable kill delay
+  @vconnection_kill_delay 120_000
+
+  # TODO: Discuss a reasonable buffer size
```

Comment:
should probably be configurable at some point but I guess 100 is an okay number for now

Reply:
Via an environment variable or via config files?

Reply:
via environment variable, similar to how (almost) everything else is configured ;)

Reply:
Ok. I'm on it.

Reply:
Can we put back `100` as discussed? Right now there is just `1`.

Reply:
@mmacai It is `1` because the resend feature is yet to be implemented/discussed. The buffer is, essentially, just stub.

Reply:
Ok.